text/0003-performance-data.md

```diff
@@ -0,0 +1,83 @@
+
+# Motivation
+[motivation]: #motivation
+
+This is needed to provide both a default and user specific implementations for future Network Profilers and Network Stats overlays. The MLAPI source code can remain Open Source, and yet, the data being streamed can be hooked up to external and low level systems.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+In order to allow first and third party rendering of the data we will need to allow both MLAPI and its transports to propagate data up the chain. 
+If the profiler flags are enabled the following will need to happen: 
+Transports will need to output the number of bytes being sent in the current tick and the latency. The data will be written out to a ProfilerTickData chunk. This data will be generic and transport agnostic.
+MLAPI will need to output the number of networked objects and the amount of RPCs being sent. This data will be added to the ProfilerTickData chunk.
+
+External Tools and APIs will be able to then get this fully written ProfilerTickData at the end of the frame for them to do what they wish (Render to Profiler Stats Overlay or Display/Record in a Profiler or file)
+
+As this data is largely a struct like object, accessing its data will be as simple as accessing fields. Additionally the rendering or logging of these statistics will fall on the user and not affect the Tick or be done at all within the MLAPI framework.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+
+In order to limit the interface, we are pushing the collected set or performance data to be obtainable only at the end of a tick. The following is roughly what this object could look like. Currently this object will be updated in ```NetworkManager.LateUpdate()``` for the MLAPI and ```Transport.Send()``` and other relevant transport method for the transport layer. At the end of the tick the user will be notified that the relevant data is complete and ready (both the MLAPI data and Transport data).
+
+```cs
+class ProfilerTickData {
+    int tickID;
+
+    // transport-level
+    int bytesSent;
+    int bytesReceived;
+    int tickDuration;
+    ...
+
+    // mlapi-level
+    int numberOfRPCs;
+    ...
+
+    // potentially? For any remaining edge cases?
+    Dictionary<string,Object> kvStore;
+}
+```
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+With a defined struct, we risk needing to know in advance as many fields and points of data that we want to push out of the Transport and MLAPI layer.
```

Comment:
I'd recommend updating the RFC to have intentional usage of struct and class. I think in this line you're really just referring to a data structure, not necessarily declaring that it is a struct or class. But if you _are_ making such a declaration then I think there should be a section about how this data is fulfilled given that structs are passed by value.